components/director/internal/tenantfetchersvc/handler.go

```diff
@@ -65,6 +89,43 @@ func NewTenantsHTTPHandler(subscriber TenantSubscriber, config HandlerConfig) *h
 	}
 }
 
+// NewTenantFetcherHTTPHandler returns a new HTTP handler, responsible for creation of on-demand tenants.
+func NewTenantFetcherHTTPHandler(fetcher TenantFetcher, config HandlerConfig) *handler {
+	return &handler{
+		fetcher: fetcher,
+		config:  config,
+	}
+}
+
+// FetchTenantOnDemand fetches CIS events for a provided subaccount and creates a tenant
```

Comment:
CIS -> External tenants registry (I'm pretty sure it's already referenced as CIS somewhere, but still) 😄